mailer/engine.py

```diff
@@ -112,6 +113,24 @@ def release_lock(lock):
     logging.debug("released.")
 
 
+def error_handler(connection, message, err):
+    error_type = type(err)
+    if error_type is socket_error or \
+        error_type is smtplib.SMTPSenderRefused or \
+        error_type is smtplib.SMTPRecipientsRefused or \
+        error_type is smtplib.SMTPDataError or \
+        error_type is smtplib.SMTPAuthenticationError:
```

Comment:
This should use `isinstance(err, (socket_error, smtplib.SMTPSenderRefused, ...))`, which would be the equivalent to the original `except (socket_error, smtplib.SMTPSenderRefused, ...)`.